source/mainarrays.f90

```diff
@@ -311,6 +315,8 @@ subroutine updatePairMap
 
   pairMap(:,:) = 0
   do j=1,npart
+    ! do not update the map in case the particle is not a primary one
+    if (pairID(1,j).eq.0.and.pairID(2,j).eq.0 ) cycle
```

Comment:
I'd suggest using modern Fortran operators `==` instead of the F77 and earlier format `.eq.`. I've been trying to clean up the old format when I modify code, since we no longer try to preserve F77 compatibility.

There should also not be a space before the closing bracket.